### PR TITLE
Add Prettier

### PR DESCRIPTION
**Pull request by rt2zz**
lets get on this bandwagon ;)

### CODE REVIEW

**Comment by marvinhagemeister** [comment]
Awesome! Prettier is a relief for the js community 💯

File: package.json [hunk 1]
```diff
@@ -17,8 +17,16 @@
     "build:umd": "cross-env BABEL_ENV=commonjs webpack",
     "build:umd:min": "cross-env BABEL_ENV=commonjs NODE_ENV=production webpack",
     "build:watch": "npm run build:commonjs -- -watch",
+    "precommit": "lint-staged",
     "prepublish": "npm run clean && npm run build"
   },
+  "lint-staged": {
```
Comment (randycoulman) [inline package.json]: I love Prettier, and using lint-staged here is super handy for contributors.  But be aware that this recipe can be very surprising if someone has used `git add -p`: any file that gets reformatted will become part of the commit, even if it was only partially staged.  See https://github.com/okonet/lint-staged/pull/75 for details.

Comment (rt2zz) [inline package.json]: do you have an alternate recommendation for how to implement prettier? I would like to ensure all code PRed is prettier'ified, wether that is via CI or lint-staged or other.

Comment (randycoulman) [inline package.json]: For now, I'm living with doing `prettier --list-different` in my pre-commit hook to fail the commit on unformatted files and then adding a `format` script to actually do the formatting. Not ideal, but better than surprising people IMO.

**Comment by rt2zz** [comment]
this is on v5 branch

**Pull request closed without merging**